docs/pages/upgrading/client-tools-autoupdate.mdx

```diff
@@ -0,0 +1,135 @@
+---
+title: Teleport Client Tools Automatic Updates
+description: Explains how to use Teleport client tools (`tsh` and `tctl`) auto-updates.
+---
+
+This documentation explains how to keep Teleport client tools like `tsh` and `tctl` up-to-date.
+Updates can be automatic or self-managed, ensuring tools are secure, free from bugs,
+and compatible with your Teleport cluster.
+
+## Why Keep Client Tools Updated?
+
+- **Security**: Updates deliver patches for known vulnerabilities.
+- **Bug Fixes**: Resolved issues are pushed to endpoints.
+- **Compatibility**: Avoid manual understanding of [Teleport component compatibility rules](https://goteleport.com/docs/upgrading/overview/#component-compatibility).
+
+---
+
+## Automatic Updates
+
+### How It Works
+
+When you run `tsh login`, the tool will check if updates are enabled for your cluster.
+If your client version differs from the cluster's required version, it will:
+
+1. Download the updated version.
+2. Store it in `~/.tsh/bin`.
+3. Validate the binary with a checksum.
+4. Re-execute using the updated version (with the same environment variables).
+
+### Key Features
```

Comment:
It seems like we can integrate this section with "How it works". Otherwise, I'm not sure what the distinction is between the two sections.

Reply:
I've added it as sub-section of `How it works` or should I just remove section definition at all, first part of how it works shows simplified steps, `key features` gives little bit more details about the process